src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.String.cs

```diff
@@ -73,47 +129,77 @@ public void WriteStringValue(ReadOnlySpan<char> value)
         {
             JsonWriterHelper.ValidateValue(value);
 
-            WriteStringEscape(value);
+            WriteStringEscape(value, JsonTokenType.String);
 
             SetFlagToAddListSeparatorBeforeNextItem();
             _tokenType = JsonTokenType.String;
         }
 
-        private void WriteStringEscape(ReadOnlySpan<char> value)
+        /// <summary>
+        /// Writes the text value segment as a partial JSON string.
+        /// </summary>
+        /// <param name="value">The value to write.</param>
+        /// <param name="isFinalSegment">Indicates that this is the final segment of the string.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the specified value is too large.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this would result in invalid JSON being written (while validation is enabled).
+        /// </exception>
+        /// <remarks>
+        /// The value is escaped before writing.
+        /// </remarks>
+        public void WriteStringValueSegment(ReadOnlySpan<char> value, bool isFinalSegment)
+        {
+            JsonWriterHelper.ValidateValue(value);
+
+            JsonTokenType nextTokenType = isFinalSegment ? JsonTokenType.String : JsonTokenType.StringSegment;
+
+            WriteStringEscape(value, nextTokenType);
```

Comment:
There needs to be some kind of provision for surrogate pairs that are split between segments.

Reply:
Would blocking be suitable (i.e. if the final character is a low surrogate)? This would push a requirement to clients to ensure they're pushing semantically complete characters but saves complexity here (about the only thing I could think of would be to remember the partial surrogate and strip it when we write the current string segment).

Reply:
> This would push a requirement to clients to ensure they're pushing semantically complete characters but saves complexity here.

Why force such a constraint? Validation seems as simple as storing the low surrogate from the previous block and not doing so would just pass that responsibility on to the caller.